src/server/scripts/BrokenIsles/BlackRookHold/boss_amalgam_of_souls.cpp

```diff
@@ -0,0 +1,639 @@
+[1] (AreaTriggerSpline) [3] Points: X: 3266.8752 Y: 7579.8774 Z: 12.884896
+[1] (AreaTriggerSpline) Computed Distance: 11.208270416945858
+[1] (AreaTriggerSpline) Computed Speed: 8.000193017091977
+[1] TargetRollPitchYaw: X: 0 Y: 0 Z: 370.0098
+[1] Radius: 3.5
+[1] RadiusTarget: 3.5
+[1] FieldFlags: 0 (None)
+[1] EntryID: 9899
+[1] DynamicFlags: 0
+[1] Scale: 1
+[1] (OverrideScaleCurve) StartTimeOffset: 0
+[1] (OverrideScaleCurve) [0] Points: X: 0 Y: 0
+[1] (OverrideScaleCurve) [1] Points: X: 0 Y: 0
+[1] (OverrideScaleCurve) ParameterCurve: 0
+[1] (OverrideScaleCurve) OverrideActive: False
+[1] Caster: Full: 0x204278BBA0603B80004AE40000069E4B Creature/0 R4254/S19172 Map: 1501 (Black Rook Hold) Entry: 98542 (Amalgam of Souls) Low: 433739
+[1] Duration: 30000
+[1] TimeToTarget: 1401
+[1] TimeToTargetScale: 30000
+[1] TimeToTargetExtraScale: 0
+[1] TimeToTargetPos: 30000
+[1] SpellID: 195254
+[1] SpellForVisuals: 195254
+[1] (SpellVisual) SpellXSpellVisualID: 90198
+[1] (SpellVisual) ScriptVisualID: 0
+[1] BoundsRadius2D: 3.5
+[1] DecalPropertiesID: 39
+[1] CreatingEffectGUID: Full: 0xBC4278BBA0BEAD83004AE40005069ECA Cast/3 R4254/S19172 Map: 1501 (Black Rook Hold) Entry: 195254 (Swirling Scythe) Low: 84319946
+[1] NumUnitsInside: 0
+[1] NumPlayersInside: 0
+[1] OrbitPathTarget: Full: 0x0
+[1] RollPitchYaw: X: 0 Y: 0 Z: 0
+[1] PositionalSoundKitID: 0
+[1] (ExtraScaleCurve) StartTimeOffset: 0
+[1] (ExtraScaleCurve) [0] Points: X: 0 Y: 0
+[1] (ExtraScaleCurve) [1] Points: X: 0 Y: 0
+[1] (ExtraScaleCurve) ParameterCurve: 1065353217
+[1] (ExtraScaleCurve) OverrideActive: True
+[1] HeightIgnoresScale: False
+[1] Field_261: False
+[1] (OverrideMoveCurveX) StartTimeOffset: 0
+[1] (OverrideMoveCurveX) [0] Points: X: 0 Y: 0
+[1] (OverrideMoveCurveX) [1] Points: X: 0 Y: 0
+[1] (OverrideMoveCurveX) ParameterCurve: 0
+[1] (OverrideMoveCurveX) OverrideActive: False
+[1] (OverrideMoveCurveY) StartTimeOffset: 0
+[1] (OverrideMoveCurveY) [0] Points: X: 0 Y: 0
+[1] (OverrideMoveCurveY) [1] Points: X: 0 Y: 0
+[1] (OverrideMoveCurveY) ParameterCurve: 0
+[1] (OverrideMoveCurveY) OverrideActive: False
+[1] (OverrideMoveCurveZ) StartTimeOffset: 0
+[1] (OverrideMoveCurveZ) [0] Points: X: 0 Y: 0
+[1] (OverrideMoveCurveZ) [1] Points: X: 0 Y: 0
+[1] (OverrideMoveCurveZ) ParameterCurve: 0
+[1] (OverrideMoveCurveZ) OverrideActive: False
+[1] (VisualAnim) AnimationDataID: 4294967295
+[1] (VisualAnim) AnimKitID: 0
+[1] (VisualAnim) AnimProgress: 0
+[1] (VisualAnim) IsDecay: False*/
+struct at_amalgam_of_souls_swirling_scythe : AreaTriggerAI
+{
+    using AreaTriggerAI::AreaTriggerAI;
+
+    void OnInitialize() override
+    {
+        Position destPos = at->GetPosition();
+        PathGenerator path(at);
+        path.CalculatePath(destPos.GetPositionX(), destPos.GetPositionY(), destPos.GetPositionZ(), false);
+
+        at->InitSplines(path.GetPath());
+    }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (!unit->IsPlayer() || !at->GetCaster())
```

Comment:
`at->GetCaster()` is unused, check relevant?

Reply:
actually no